python/dolfinx/nls.py

```diff
@@ -0,0 +1,58 @@
+# Copyright (C) 2021 Jørgen S. Dokken
+#
+# This file is part of DOLFINx (https://www.fenicsproject.org)
+#
+# SPDX-License-Identifier:    LGPL-3.0-or-later
+"""Methods for solving nonlinear equations."""
+
+import types
+from petsc4py import PETSc
```

Comment:
Do we need to import if we only want type hints?

Reply:
It will be possible from python 3.10: https://docs.python.org/pt-br/3.10/whatsnew/3.10.html#pep-613-typealias